odoo/orm/table_objects.py

```diff
@@ -167,6 +167,10 @@ def apply_to_database(self, model: BaseModel):
             definition_clause = self._index_definition(model.pool)
         else:
             definition_clause = self._index_definition
+        if not definition_clause:
+            # Don't create index with an empty definition
+            return
```

Comment:
I think we may also fix/ref the `get_definition` since the `definition_clause` can be empty str.